bin/oneoff/deprecate_unused_poste_urls.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+
+require 'sequel'
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require src_dir 'database'
+require_relative './poste_urls_constants'
+
+# This script is a dry-run unless we change DRY_RUN to false
+DRY_RUN = true
+
+if DRY_RUN
+  puts "DRY RUN..."
+else
+  puts "Starting update..."
+end
+
+DB.transaction do
+  # mark all poste_urls that are not in urls_to_keep as deleted_at = now
+  # this will take a long time to run as it iterates over the entire poste_urls table
+  urls_kept = []
+  deleted_at_time = DateTime.now
+  DB[:poste_urls].each do |poste_url|
```

Comment:
Not sure if this is strictly necessary, but will this load 3.5 million rows into memory? I've seen something like this used in other similar cases:

https://www.rubydoc.info/github/jeremyevans/sequel/Sequel%2FDataset:paged_each

Reply:
that's a good point, I'll update to use that method

Reply:
discussed offline and switched to different update strategy